packages/audio-player-ui/src/hooks/useAudioPlayer.ts

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+
+export function useAudioPlayer() {
+  const [currentTrack, setCurrentTrack] = useState<HTMLAudioElement | null>(
+    null
+  );
+  const [duration, setDuration] = useState(0);
+  const [curTime, setCurTime] = useState(0);
+  const [playing, setPlaying] = useState(false);
+  const [clickedTime, setClickedTime] = useState(0);
+  const [muted, setMuted] = useState(false);
+  console.log("currentTrack", currentTrack);
+
+  const toggleMute = () => {
```

Comment:
We should memoize all the handler functions with react useCallback
https://dmitripavlutin.com/dont-overuse-react-usecallback/

example from similar set of handlers:
https://github.com/dblodorn/blocksyncer/blob/main/%40media-player/useMediaPlayer.tsx

borrow as much as you like - and or - fork this repo and mess around with the @media-player directory. It was written to eventually be a plugin.